lib/sentry/sources.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Sentry.Sources do
+  @moduledoc """
+  This module is responsible for providing functionality that stores
+  the text of source files during compilation for displaying the
+  source code that caused an exception.
+
+  ### Configuration
+  There is configuration required to set up this functionality.  The options
+  include `:enable_source_code_context`, `:root_source_code_path`, `:context_lines`,
+  `:exclude_patterns`, and `:source_code_path_pattern`.
+
+  * `:enable_source_code_context` - when `true`, enables reporting source code
+    alongside exceptions.
+  * `:root_source_code_path` - The path from which to start recursively reading files from.
+    Should usually be set to `File.cwd!`.
+  * `:context_lines` - The number of lines of source code before and after the line that
+    caused the exception to be included.  Defaults to `3`.
+  * `:exclude_patterns` - a list of Regex expressions used to exclude file paths that
+    should not be stored or referenced when reporting exceptions.  Defaults to
+    `[~r"/_build/", ~r"/deps/", ~r"/priv/"]`.
+  * `:source_code_path_pattern` - a glob that is expanded to select files from the
+    `:root_source_code_path`.  Defaults to `"**/*.ex"`.
+
+  An example configuration:
+
+      config :sentry,
+        dsn: "https://public:secret@app.getsentry.com/1",
+        root_source_code_path: File.cwd!,
+        context_lines: 5
+
+  ### Source code storage
+
+  The file contents are saved when Sentry is compiled, which can cause some
+  complications.  If a file is changed, and Sentry is not recompiled,
+  it will still report old source code.
+
+  The best way to ensure source code is up to date is to recompile Sentry
```

Comment:
This probably needs to also be apart of main docs, "on deployment it is advised to recompile sentry to make sure you are always reporting the latest code."